### PR TITLE
Loop as long as with duration limit

### PR DESCRIPTION
**Pull request by wiacekm**
I think it would be good to add another type of loop statement where we iterate over the loop as long as condition is satisfied but no longer than specified amount of time.
It would allow to avoid some tricky solution and hacks when we have to wait for some data in response. And we cannot predict when it will be populated (ex due to caching).

### CODE REVIEW

File: gatling-core/src/main/scala/io/gatling/core/action/builder/LoopBuilder.scala.orig [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+case object DoWhileType extends LoopType("doWhile", false, true)
+=======
+sealed abstract class LoopType(val name: String, val timeBased: Boolean)
+case object RepeatLoopType extends LoopType("repeat", false)
+case object ForeachLoopType extends LoopType("foreach", false)
+case object DuringLoopType extends LoopType("during", true)
+case object ForeverLoopType extends LoopType("forever", false)
+case object AsLongAsLoopType extends LoopType("asLongAs", false)
+case object AsLongAsDuringLoopType extends LoopType("asLongAsDuring", true)
+>>>>>>> Stashed changes
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/action/builder/LoopBuilder.scala.orig]: Broken merge conflict.

**Comment by wiacekm** [comment]
hopefully everything is ok now

**Comment by wiacekm** [comment]
let me know if I should open new pull request or reopen this one

**Comment by slandelle** [comment]
cherry-picked, thanks!

**Pull request closed without merging**